infra/base-images/base-builder/compile_go_fuzzer

```diff
@@ -23,9 +23,14 @@ if [[ $#  -eq 4 ]]; then
   tags="-tags $4"
 fi
 
+cd $GOPATH/src/$path || true
+# project was downloaded with go get if go list fails
+go list $path || cd $GOPATH/pkg/mod/ && cd `echo $path | cut -d/ -f1-3 | awk '{print $1"*"}'`
```

Comment:
if `go list` fails, we are not in some good directory, try module cache (where `go get` puts them since go 1.16)

Reply:
I don't think you should be guessing paths inside the module cache. In the original issue I showed a way to do this with `go list -f {{.Dir}}`

Reply:
@catenacyber - can you please do this in another PR. for now, merging this.

Reply:
> I don't think you should be guessing paths inside the module cache

Indeed I should not, but `go list -m -f {{.Dir}}` needs network connectivity...
May it return a newer version than the one downloaded with `go get` when building the dockerfile ?